packages/ocvImpact/R/utils.R

```diff
@@ -181,6 +182,327 @@ generate_pct_protect_function <- function(my_trunc_year = 5, my_ve_scen = "base"
+#' @importFrom magrittr %>%
+#' @return dataframe with estimates for unweighted and weighted mean vaccination campaign coverage proportion across coverage surveys in the review
+#' @export
+generate_pct_protect_function_two_dose_under5 <- function(my_trunc_year = 5, my_ve_scen = "base", study_type = "efficacy"){
+  
+  ve.dat <- readxl::read_xlsx("input_data/SR_of_OCV_efficacy_effectiveness.xlsx", sheet = 8) #load review table data from Hanmeng&Andrew
+  
+  ve.dat <- ve.dat[ve.dat$Dose == 2,] #select only cases with two doses of the vaccine, ignore cases with "at least one dose" for now
+  
+  if (study_type == "efficacy"){
+    ve.dat <- ve.dat[ve.dat$StudyType == 'Efficacy',]
+  } else if (study_type == "effectiveness"){
+    ve.dat <- ve.dat[ve.dat$StudyType == 'Effectiveness',]
+  } else if (study_type == "all"){
+    ve.dat <- ve.dat
+  }
+  
+  ve.dat$T <- (ve.dat$TL+ve.dat$TR)/2
+  
+  ve.dat$weights <-  1/(abs(as.numeric(ve.dat$se))^2)
+  
+  ve.dat <- ve.dat[ve.dat$TL<48,] #select only cases where the observation period ended before two years 
+  
+  ##this is our basic trend in vaccine effectiveness.
+  ve.trend <- lm(yi~T , data=ve.dat, weights = weights)
+  
+  ##create a function that gives the expected percent protected
+  ##by a vaccine during a particular year after vaccination
+  
+  pct.protect<-function(year, ci=FALSE,trunc_year=my_trunc_year,ve_scen=my_ve_scen) {
+    if (any(year%%1 !=0)) {warning("function designed to average across years only")}
+    if(!ve_scen %in% c("base","low","high")) {warning("ve_scen not recognized, assuming ve_scen='base'")}
+    
+    months <- (year-.5)*12
+    
+    if(ve_scen %in% c("high","low")){
+      ci <- TRUE
+    }
+    
+    if (ci) {
+      rc <- pmax(1-exp(predict(ve.trend, newdata=data.frame(T=months), interval="confidence") ),0) %>% unname
+    } else {
+      rc <- pmax(1-exp(predict(ve.trend, newdata=data.frame(T=months))),0) %>% unname
+    }
+    
+    rc <- as.matrix(rc)
+    rc <- rc-(rc*0.44) ##added to shift down the curve by 0.44 of the vaccine efficacy for over 5 year olds 
+    
+    if(any(year>trunc_year)){
+      rc[which(year>trunc_year),] <- 0
+    }
+    
+    if(ve_scen == "high"){
+      rc <- rc[,2]
+    } else if (ve_scen == "low"){
+      rc <- rc[,3]
+    }
+    
+    return(rc)
+  }
+  return(pct.protect)
+}
+
+
+
+#' @title vaccine_efficacy_all_groups_one_dose
+#' @description wrapper function that takes the vaccine efficacy functions for the <5 and >5 year olds for the one-dose campaign and returns a new function that returns their weighted average. Weighting is according to the proportion of <5 and >5 year-olds in the population
+#' @return a function returning the weighted average vaccine efficacy for the one-dose campaign 
+#' @export
+#'
+#' @examples
+vaccine_efficacy_all_groups_one_dose <- function(){
+  vaccine_efficacy <- function(proportion_under5, years){
+    vaccine_efficavy_under5_function <- generate_pct_protect_function_one_dose_under5()
```

Comment:
@christinaalam Can you correct the spellying typos for  `efficacy`?

Reply:
corrected!